rat/fit.py

```diff
@@ -31,6 +43,21 @@ def _read_folder_into_dict(folder: str):
     return dfs
 
 
+def _build_constrained_df(constrained_variable: numpy.array, base_df: pandas.DataFrame):
```

Comment:
Looks like we can create a post-processing class that takes a `Fit` object that would evaluate any generated quantities and then create a single dataframe.

I dunno. I like the idea of having some kind of "toolchain" that takes some object as input, does only whatever it is designed to do, and then pass it into the next class in the queue instead of one giant class that does everything inside.

Reply:
Good point I think I agree. I think breaking the sample/optimize functions out from the model class was a good idea because it makes what happens in any given class simpler.

Even though optimize and sample are only used in one place, it was more confusing baking them into the model.

I guess this is what they call coding to interfaces.